hicommonwealth/commonwealth
# User-Onboarding: User Training Card

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/7194

## Description of Changes
Added User training slider for newly onboarded users. The cards on the slider get marked as completed if the user already completed those actions or as the user performs those specified actions. 

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_USER_ONBOARDING_ENABLED`
- Create a new account and complete the welcome onboard modal steps
- Verify you see the `Welcome to Common` slider with cards (this is the user training cards/slider)
- Verify that the cards are responsive and match [figma](https://www.figma.com/file/I5jbkBPzNrbYMHYQJb6CNY/Onboarding?type=design&node-id=802-4043&mode=design&t=KtStYQfUcTMEjnth-4)
- Verify each card functions properly and is properly marked completed
  - `Give Upvotes card`: Verify that upvoting any comment/thread in any community marks this card as completed, and the next time you hard reload the page, this card is not displayed
  - `Create Content card`: Verify that creating any comment/thread in any community marks this card as completed, and the next time you hard reload the page, this card is not displayed
  - `Finish Profile card`: Verify that editing the profile from `/profile/edit` and adding social links marks this card as completed, and the next time you hard reload the page, this card is not displayed. If during profile creation you provided an email, also then this card is marked as completed.
  - `Explore Communities card`: Verify that visiting the `/communities` from the cta button of this card, marks this card as completed, and the next time you hard reload the page, this card is not displayed
- Verify that clicking on the `x` button on any card dismisses it permanently (even after page reload)
- Verify that clicking on the `Dismiss all` button, dismisses all cards permanently (even after the page reload)
- Verify that the slider is only shown to logged-in users
- Verify that when you switch to a different account, the slider state reflects the training process of that specific user account

Note: 
- The permanent dismissal state is stored in the local storage key -> `user-training-cards-store`, for testing it's easier to clear this state and reload.
- We are not doing negative checks in real-time. If the user completes a specified action (like upvoting any comment/thread) and then at a later time, removes the upvote, the training card will still be marked completed. Now if the user clears their local storage and then reloads the page (where we get fresh data from API and we don't have markers in local storage that indicate the user completing those actions), they will see the specified action as not completed again.

## Deployment Plan
N/A

## Other Considerations
N/A
